### PR TITLE
csi: No longer install the VolumeReplication CRDs from Rook

### LINKED ISSUE #8573: optional volume replication crds in helm chart
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

Make the `openshift` `crds` optional in the helm chart.

**What is use case behind this feature:**

I run netapp `trident` in the same cluster with `rook` and `trident` currently checks for any `openshift` crds to alter it's installation process...`scc` or not etc.

**Environment**:

Running `rook` in a non-`openshift` environment in conjunction with netapp `trident`.

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The volume replication CRDs are an external component, not owned by Rook. Therefore, they should be installed as any other independent component in case the admin will install other consumers of the volumereplication CRDs in the future in addition to Rook and the CSI driver.

The volume replication controller in the Ceph-CSI driver was already optional and disabled by default. To enable the volume replication controller there is just now an additional step to install the CRDs if the feature is to be enabled.

Since the CRDs are no longer included with Rook, downstream may need to find a new integration path for the CRDs.

**Which issue is resolved by this Pull Request:**
Resolves #8573 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-csi-drivers.md
Comment (Madhu-1) [reply Documentation/ceph-csi-drivers.md:88]: the main might have the latest/breaking changes. using the release branch should be good. and the release should match the image tag which is used in operator.yaml or what comes with default. For example if the image tag is v0.1.0 the CRD's should be installed from the same tag

File: tests/integration/ceph_smoke_test.go
Comment (sp98) [inline tests/integration/ceph_smoke_test.go:101]: why remove this check? We still support lower versions, right?

**Review by leseb — CHANGES_REQUESTED** [review]
You need to remove `build/crds/crds.go` too, which should remove the module dep too.

File: tests/framework/installer/settings.go
Comment (leseb) [inline tests/framework/installer/settings.go:49]: Is this change fixing anything?

Comment (travisn) [reply tests/framework/installer/settings.go:49]: A new method `readManifestFromURL()` was added to load from a different url than the rook github.

File: Documentation/ceph-csi-drivers.md
Comment (travisn) [reply Documentation/ceph-csi-drivers.md:88]: Ok, for now I'll use v0.1.0. Looks like v0.2.0 is out, so we will probably update to that version soon.

File: tests/integration/ceph_smoke_test.go
Comment (sp98) [reply tests/integration/ceph_smoke_test.go:101]: ok. How about the same check removed in `ceph_object_test.go` file above?

File: Documentation/ceph-csi-drivers.md
Comment (Madhu-1) [reply Documentation/ceph-csi-drivers.md:88]: Yes, we can update that one. Do you want to take care of that in this PR. or can I send separate PR for that one?

Comment (travisn) [reply Documentation/ceph-csi-drivers.md:88]: Separate PR sounds good

Comment (Madhu-1) [reply Documentation/ceph-csi-drivers.md:88]: Will send it once this PR is merged :+1:

**Review by sp98 — APPROVED** [review]

File: images/ceph/Makefile
Comment (leseb) [inline images/ceph/Makefile:115]:
```suggestion
	curl -L $(VOL_REPL_URL)/$(VOLUME_REPLICATION_CRD) -o $(CURL_TEMPLATE_DIR)/$(VOLUME_REPLICATION_CRD)
```

Typically safer if we get a redirect.

Comment (leseb) [reply images/ceph/Makefile:115]: Where is. `CURL_TEMPLATE_DIR` coming from?

**Review by leseb — CHANGES_REQUESTED** [review]
As discussed, waiting for https://github.com/rook/rook/pull/8862

**Review by leseb — APPROVED** [review]
Waiting for https://github.com/rook/rook/pull/8862 is a nice to have but shouldn't block the release.

**Pull request merged**
